### PR TITLE
matlab example

### PR DESCRIPTION
**Pull request by jburel**
Example for neubias training

cc @pwalczysko

### CODE REVIEW

File: practical/matlab/analyze_dataset_save_rois_and_attachments.m [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    [B,L] = bwboundaries(BWnobord, 'noholes');
+    roi = omero.model.RoiI;
+    for k = 1 : length(B)
+       boundary = B{k};
+       x_coordinates = boundary(:,2);
+       y_coordinates = boundary(:,1);
+       shape = createPolyline(x_coordinates, y_coordinates);
+       roi.addShape(shape);
+       area = polyarea(x_coordinates, y_coordinates);
+       max_area = max(max_area, area);
```
[the remark this answers is not in the document]
Comment (pwalczysko) [reply practical/matlab/analyze_dataset_save_rois_and_attachments.m]: Yes, so it is this line, and the problem is that after the https://github.com/ome/training-scripts/pull/41/files#diff-de6aadc1d2f78d4f8fd518a1dbbc7528R75 line, there is no definition of ``max_area``. See https://github.com/ome/training-scripts/pull/41/files#diff-546de7b9022a65150237a305bafe382aR81 - I added this statement to the present script here, and it started working, cc @jburel

**Comment by pwalczysko** [comment]
When the first script in the diff is run against idr0021 images, the resulting ROIs do not match the Fiji segmentation ROIs, they seem to not to correspond to the interesting features at all.

screen shot 2018-09-24 at 19 42 00

The ROIs seem to be polylines in OMERO, not polygons. I think we want polygons.

screen shot 2018-09-24 at 19 42 10

**Comment by pwalczysko** [comment]
For the first script, I have

  - 1 bug https://github.com/ome/training-scripts/pull/41#discussion_r219944415
  - 2 RFEs (improve parameters to detect good features on idr0021 and use polygon) https://github.com/ome/training-scripts/pull/41#issuecomment-424082372

I will continue with the other 2 scripts tomorrow

**Comment by jburel** [comment]
@pwalczysko hold on I cannot remember where I left the scripts.
I will review first

**Comment by jburel** [comment]
@pwalczysko the scripts should now be working. For the algo, I did not focus on the accuracy. I mainly put in place the structure so we can have all the pieces together.

**Comment by pwalczysko** [comment]
The first script works fine now. But the shapes are not channel-associated.

File: practical/matlab/idr0021.m [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+% Detect cells using image segmentation
+% see https://www.mathworks.com/examples/image/mw/images-ex64621327-detecting-a-cell-using-image-segmentation
+% The shapes are saved as polylines.
+% The script has been tested with Matlab2017a
+
+host='outreach.openmicroscopy.org';
+% To be modified
+user='USERNAME';
+password='PASSWORD';
+password='ome';
```
[the remark this answers is not in the document]
Comment (pwalczysko) [reply practical/matlab/idr0021.m]: The script does not work if this line is left in. Looks like omission

**Comment by pwalczysko** [comment]
The second script works except for the bug https://github.com/ome/training-scripts/pull/41#discussion_r220123968. But the resulting table on the idr0021 project is not usable in parade - parade performs the middle-pane-vanishes trick when Table >  Area filter is selected and the slider is dragged

**Comment by pwalczysko** [comment]
The FRAP script works, but

   - it crashes if there are any images with no ROIs in the dataset (all images MUST have ROIs - this seems to me not reasonable, if some images have ROIs, work with them, if not, leave such images alone)
   - it produces only the pngs of the plots attached to the images, no values for further analysis, correct ? I think we should get at least a CSV yet, otherwise how would you do any follow-up analysis ?

**Comment by jburel** [comment]
@pwalczysko parade expects an Image column, and the script created a table with ImageID

**Comment by jburel** [comment]
@pwalczysko the frap script creates a map annotation, you probably missed it. I opted for that option since the other scripts create CSV and table

**Comment by jburel** [comment]
@pwalczysko I fixed the problem with table and shapes are now attached to a given channel

**Comment by pwalczysko** [comment]
Re: The script here which runs on idr0021 project:

   - the resulting CSV ``results.csv``, see screenshot below, not sure what results are in the sheet, is that just one image per Dataset, the one with the biggest area maybe ? I think it could be explained in the column titles.

screen shot 2018-09-26 at 16 08 25

Is the area in pixels ?

**Comment by pwalczysko** [comment]
In summary for the idr0021 scipt:
   - issue 1 https://github.com/ome/training-scripts/pull/41#issuecomment-424752955 (just one image per dataset has metadata from matlab)
   - issue 2 https://github.com/ome/training-scripts/pull/41#issuecomment-424756739 (Table_Dataset_name is of a wrong format in a wrong place)

The ROIs are associated with a channel now.

**Comment by pwalczysko** [comment]
In summary for FRAP script:

   - all good, selects just images with ROIs, produces KVPs and attaches pngs

**Comment by pwalczysko** [comment]
The script which works on a Dataset is fine.

Total summary: out of 3 scripts, the Dataset one is fine, the idr0021 and FRAP scripts have problems https://github.com/ome/training-scripts/pull/41#issuecomment-424760854 and https://github.com/ome/training-scripts/pull/41#issuecomment-424764532

**Comment by jburel** [comment]
@pwalczysko the idr0021 should now work. I have also removed the dataset's name from the table
What is the problem with FRAP? you indicated in the office that it works

**Comment by pwalczysko** [comment]
https://github.com/ome/training-scripts/pull/41/commits/f7aa15a5f3944823567bacdd434b6f4b77360c5c and https://github.com/ome/training-scripts/pull/41/commits/c8c80a2ca66d1872ded3eda3bba4bce37d149494 commits have the desired effect - all the images are now included, and the dataset name is not in the table.

**Comment by pwalczysko** [comment]
There is a problem with the FRAP CSV. Although it exists and is attached correctly, the values inside are not right (are all the same for all timepoints)

screen shot 2018-09-28 at 16 49 43

**Comment by jburel** [comment]
@pwalczysko sorry I forgot to correct it in the script when I copy/paste from matlab editor

**Comment by pwalczysko** [comment]
Yes, the values in the CSV are now as expected.

Ready to merge FMPOV

**Comment by pwalczysko** [comment]
Sorry, maybe we need to spend some time on optimizing the ROI parameters as discussed. We can do it in a follow-up PR ?

**Comment by jburel** [comment]
@pwalczysko no rush to merge the PR. We have the pieces we wanted in place. We can look at parameters now

**Comment by pwalczysko** [comment]
I think this all (all three scripts) work satisfactorily now. Quickly retested also the ``dataset`` script and it seems to deliver the same results as the ``idr0021`` script.

Ready to merge FMPOV.

**Comment by jburel** [comment]
@pwalczysko do you want to have a final look?

**Comment by pwalczysko** [comment]
screen shot 2018-10-08 at 15 41 37


With idr0021 script, the plot did not work out. See above.

**Comment by pwalczysko** [comment]
RFE: 

for the script which is run on one dataset:

   - when "false" is passed (Graeme's algo used), all is perfect
   - when "true" is passed, I do not have the plot in matlab, I would think I should get one too, such as I did when "false" was passed above ?

**Comment by pwalczysko** [comment]
Except for the RFE https://github.com/ome/training-scripts/pull/41#issuecomment-428120495 there is no problem with either the ``dataset`` script or the ``idr0021`` script. Although the ``idr0021`` script does not produce the graph in Matlab, I think this is actually valid and what we want.
There were no changes in the FRAP script since last testing.

Ready to merge FMPOV

**Comment by pwalczysko** [comment]
The https://github.com/ome/training-scripts/pull/41#issuecomment-428120495 is not valid. The window is there.

All good here

**Pull request merged**